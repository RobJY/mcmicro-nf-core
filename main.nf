#!/usr/bin/env nextflow
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    nf-core/mcmicro
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Github : https://github.com/nf-core/mcmicro
    Website: https://nf-co.re/mcmicro
    Slack  : https://nfcore.slack.com/channels/mcmicro
----------------------------------------------------------------------------------------
*/

nextflow.enable.dsl = 2

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    IMPORT FUNCTIONS / MODULES / SUBWORKFLOWS / WORKFLOWS
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

include { MCMICRO  } from './workflows/mcmicro'
include { PIPELINE_INITIALISATION } from './subworkflows/local/utils_nfcore_mcmicro_pipeline'
include { PIPELINE_COMPLETION     } from './subworkflows/local/utils_nfcore_mcmicro_pipeline'

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    NAMED WORKFLOWS FOR PIPELINE
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

//
// WORKFLOW: Run main analysis pipeline depending on type of input
//
workflow NFCORE_MCMICRO {

    take:
    samplesheet // channel: samplesheet read in from --input

    main:

    //
    // WORKFLOW: Run pipeline
    //
    MCMICRO (
        samplesheet
    )

    emit:
    multiqc_report = MCMICRO.out.multiqc_report // channel: /path/to/multiqc_report.html

}
/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    RUN MAIN WORKFLOW
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/

workflow {

    main:

    //
    // SUBWORKFLOW: Run initialisation tasks
    //
    PIPELINE_INITIALISATION (
        params.version,
        params.help,
        params.validate_params,
        params.monochrome_logs,
        args,
        params.outdir,
        params.input
    )

    //
    // WORKFLOW: Run main workflow
    //
    NFCORE_MCMICRO (
        PIPELINE_INITIALISATION.out.samplesheet
    )

    //
    // SUBWORKFLOW: Run completion tasks
    //
    PIPELINE_COMPLETION (
        params.email,
        params.email_on_fail,
        params.plaintext_email,
        params.outdir,
        params.monochrome_logs,
        params.hook_url,
        NFCORE_MCMICRO.out.multiqc_report
    )
}

/*
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    THE END
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*/
