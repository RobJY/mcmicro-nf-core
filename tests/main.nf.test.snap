{
    "sample: multiple cycle ashlar input, no illumination correction": {
        "content": [
            "TEST1.ome.tif:md5,9d4fca1d7ec97be3f6343eb712af9dea",
            "mask_TEST1.tif:md5,c1075eb025f558fcadedc1f0d903ec15",
            "marker_sheet.valid.csv:md5,93b58772a86c6eb122809f5fc29d1a66",
            "TEST1_mask_TEST1.csv:rounded:md5,5b5036130a84dc31268007aaf3322917"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T16:02:12.196826937"
    },
    "sample: multiple cycle ashlar input, basicpy illumination correction": {
        "content": [
            {
                "format": "OME-TIFF",
                "type": "uint16",
                "sizeX": 591,
                "sizeY": 470,
                "sizeZ": 1,
                "sizeC": 8,
                "sizeT": 1,
                "physicalSizeX": 0.65,
                "physicalSizeY": 0.65,
                "physicalSizeZ": null
            },
            "mask_TEST1.tif:md5,b4d705f488f83e1201e61c353268b395",
            "marker_sheet.valid.csv:md5,93b58772a86c6eb122809f5fc29d1a66",
            "TEST1_mask_TEST1.csv:rounded:md5,51187836ffeeabfee32ab5273e701b3f",
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T16:02:39.433809986"
    },
    "cycle: no illumination correction; spaces in markersheet (ticket #22)": {
        "content": [
            "TEST1.ome.tif:md5,ae98ac74e3c818bb26c99ba4b9c2dd51",
            "mask_TEST1.tif:md5,c1075eb025f558fcadedc1f0d903ec15",
            "marker_sheet.valid.csv:md5,15fa6d74598c89fd01abfafa9822144d",
            "TEST1_mask_TEST1.csv:rounded:md5,ea78917fc9b21e5b6f424627e6226da1"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T15:58:57.545086596"
    },
    "cycle: multiple file ashlar input no correction": {
        "content": [
            "cycif-tonsil.ome.tif:md5,5f4165203545b7efe04ed5f4a6d5ac9d",
            "mask_cycif-tonsil.tif:md5,c1075eb025f558fcadedc1f0d903ec15",
            "marker_sheet.valid.csv:md5,8b59a59f92c40da7a129a32e04458f9a",
            "cycif-tonsil_mask_cycif-tonsil.csv:rounded:md5,01286b0d3b0245d5ece0d279c0a5391e"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T16:00:36.270528511"
    },
    "cycle: basicpy illumination correction": {
        "content": [
            {
                "format": "OME-TIFF",
                "type": "uint16",
                "sizeX": 591,
                "sizeY": 470,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 0.65,
                "physicalSizeY": 0.65,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "int32",
                "sizeX": 591,
                "sizeY": 470,
                "sizeZ": 1,
                "sizeC": 1,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            "marker_sheet.valid.csv:md5,f1cf3d8e7d8fe5134c698f77fb29ae25",
            "TEST1_mask_TEST1.csv:rounded:md5,9459567cdd2a3c49688278f5d5fed998",
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T16:23:17.178865787"
    },
    "cycle: multiple file ashlar input with basicpy correction": {
        "content": [
            {
                "format": "OME-TIFF",
                "type": "uint16",
                "sizeX": 591,
                "sizeY": 470,
                "sizeZ": 1,
                "sizeC": 12,
                "sizeT": 1,
                "physicalSizeX": 0.65,
                "physicalSizeY": 0.65,
                "physicalSizeZ": null
            },
            "mask_cycif-tonsil.tif:md5,b4d705f488f83e1201e61c353268b395",
            "marker_sheet.valid.csv:md5,8b59a59f92c40da7a129a32e04458f9a",
            "cycif-tonsil_mask_cycif-tonsil.csv:rounded:md5,0e34fe3038581b06395bd04546d17b3b",
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T16:01:24.224272296"
    },
    "sample: no illumination correction": {
        "content": [
            "TEST1.ome.tif:md5,ae98ac74e3c818bb26c99ba4b9c2dd51",
            "mask_TEST1.tif:md5,c1075eb025f558fcadedc1f0d903ec15",
            "marker_sheet.valid.csv:md5,f1cf3d8e7d8fe5134c698f77fb29ae25",
            "TEST1_mask_TEST1.csv:rounded:md5,a7c226a3d32a94dd05356c5998eae522"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T15:59:59.693635237"
    },
    "cycle: multiple file ashlar input with multiple samples and basicpy correction": {
        "content": [
            {
                "format": "OME-TIFF",
                "type": "uint16",
                "sizeX": 591,
                "sizeY": 470,
                "sizeZ": 1,
                "sizeC": 8,
                "sizeT": 1,
                "physicalSizeX": 0.65,
                "physicalSizeY": 0.65,
                "physicalSizeZ": null
            },
            {
                "format": "OME-TIFF",
                "type": "uint16",
                "sizeX": 591,
                "sizeY": 468,
                "sizeZ": 1,
                "sizeC": 8,
                "sizeT": 1,
                "physicalSizeX": 0.65,
                "physicalSizeY": 0.65,
                "physicalSizeZ": null
            },
            "mask_cycif-tonsil.tif:md5,b4d705f488f83e1201e61c353268b395",
            "mask_cycif-tonsil2.tif:md5,bacaeced4f2159b8b7f3b42d173f995c",
            "marker_sheet.valid.csv:md5,93b58772a86c6eb122809f5fc29d1a66",
            "cycif-tonsil_mask_cycif-tonsil.csv:rounded:md5,51187836ffeeabfee32ab5273e701b3f",
            "cycif-tonsil2_mask_cycif-tonsil2.csv:rounded:md5,a571ff4029cd0f0caf20fd682aacd4f0",
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T16:01:53.602122612"
    },
    "cycle: multiple file ashlar input with multiple samples no correction": {
        "content": [
            "cycif-tonsil.ome.tif:md5,9d4fca1d7ec97be3f6343eb712af9dea",
            "cycif-tonsil2.ome.tif:md5,ea772fef41d398ffd76cbb6354266678",
            "mask_cycif-tonsil.tif:md5,c1075eb025f558fcadedc1f0d903ec15",
            "mask_cycif-tonsil2.tif:md5,bacaeced4f2159b8b7f3b42d173f995c",
            "marker_sheet.valid.csv:md5,93b58772a86c6eb122809f5fc29d1a66",
            "cycif-tonsil_mask_cycif-tonsil.csv:rounded:md5,5b5036130a84dc31268007aaf3322917",
            "cycif-tonsil2_mask_cycif-tonsil2.csv:rounded:md5,61eb9b30399fb29041f2d080181b6e47"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T16:00:55.775911164"
    },
    "sample: manual illumination correction": {
        "content": [
            "TEST1.ome.tif:md5,ae98ac74e3c818bb26c99ba4b9c2dd51",
            "mask_TEST1.tif:md5,c1075eb025f558fcadedc1f0d903ec15",
            "marker_sheet.valid.csv:md5,f1cf3d8e7d8fe5134c698f77fb29ae25",
            "TEST1_mask_TEST1.csv:rounded:md5,a7c226a3d32a94dd05356c5998eae522"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T16:00:17.163342084"
    },
    "cycle: no illumination correction": {
        "content": [
            "TEST1.ome.tif:md5,ae98ac74e3c818bb26c99ba4b9c2dd51",
            "mask_TEST1.tif:md5,c1075eb025f558fcadedc1f0d903ec15",
            "marker_sheet.valid.csv:md5,f1cf3d8e7d8fe5134c698f77fb29ae25",
            "TEST1_mask_TEST1.csv:rounded:md5,a7c226a3d32a94dd05356c5998eae522"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T15:58:40.245133922"
    },
    "sample: multiple cycle ashlar input with multiple samples and basicpy illumination correction": {
        "content": [
            {
                "format": "OME-TIFF",
                "type": "uint16",
                "sizeX": 591,
                "sizeY": 470,
                "sizeZ": 1,
                "sizeC": 8,
                "sizeT": 1,
                "physicalSizeX": 0.65,
                "physicalSizeY": 0.65,
                "physicalSizeZ": null
            },
            {
                "format": "OME-TIFF",
                "type": "uint16",
                "sizeX": 594,
                "sizeY": 475,
                "sizeZ": 1,
                "sizeC": 8,
                "sizeT": 1,
                "physicalSizeX": 0.65,
                "physicalSizeY": 0.65,
                "physicalSizeZ": null
            },
            "mask_TEST1.tif:md5,b4d705f488f83e1201e61c353268b395",
            "mask_TEST2.tif:md5,74da1dd8bfe02932d983087a96cd249f",
            "marker_sheet.valid.csv:md5,93b58772a86c6eb122809f5fc29d1a66",
            "TEST1_mask_TEST1.csv:rounded:md5,51187836ffeeabfee32ab5273e701b3f",
            "TEST2_mask_TEST2.csv:rounded:md5,3dc9aa2ca7676566237ac144c5ca653c",
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            },
            {
                "format": "Tagged Image File Format",
                "type": "float",
                "sizeX": 220,
                "sizeY": 180,
                "sizeZ": 1,
                "sizeC": 4,
                "sizeT": 1,
                "physicalSizeX": 1.0,
                "physicalSizeY": 1.0,
                "physicalSizeZ": null
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T16:03:08.945600735"
    },
    "cycle: manual illumination correction": {
        "content": [
            "TEST1.ome.tif:md5,ae98ac74e3c818bb26c99ba4b9c2dd51",
            "mask_TEST1.tif:md5,c1075eb025f558fcadedc1f0d903ec15",
            "marker_sheet.valid.csv:md5,f1cf3d8e7d8fe5134c698f77fb29ae25",
            "TEST1_mask_TEST1.csv:rounded:md5,a7c226a3d32a94dd05356c5998eae522"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-22T15:59:42.197064018"
    }
}